jupyterlab/handlers/ydoc_handler.py

```diff
@@ -12,33 +13,48 @@
 from tornado import web
 from tornado.websocket import WebSocketHandler
 from ypy_websocket.websocket_server import WebsocketServer, YRoom
+from ypy_websocket.ystore import BaseYStore, SQLiteYStore, TempFileYStore, YDocNotFound
 
 YFILE = YDOCS["file"]
-
 RENAME_SESSION = 127
 
 
+class JupyterTempFileYStore(TempFileYStore):
+    prefix_dir = "jupyter_ystore_"
+
+
+class JupyterSQLiteYStore(SQLiteYStore):
+    db_path = ".jupyter_ystore.db"
```

Comment:
Where is this file located? Is there one per directory? One per server? What are the tradeoffs of the two options?

Reply:
This file is located in the directory where JupyterLab (i.e. the server) is run. There is one file for the whole duration of the server, and if the server is run again in the same directory, the file is reused, so there is persistence between "JupyterLab sessions".
Having one file per directory could be interesting because we wouldn't identify a document with its path, but only its name. That means if JupyterLab is launched from another directory, we are still able to easily find the Y store for any opened document. With the current approach, the (relative) path to a document is encoded in the database, so changing the JupyterLab directory would create a new Y store and updates from the previous session would be lost.
Thanks for raising that up @ellisonbg, I think we should have one `SQLiteYStore` per directory, what do you think?